reconcile/utils/terrascript_aws_client.py

```diff
@@ -1543,38 +1543,48 @@ def populate_resources(self, ocm_map: OCMMap | None = None) -> None:
             for spec in specs:
                 self.populate_tf_resources(spec, ocm_map=ocm_map)
 
-    def _get_provisioner_provider_exclusions(
+    def _is_provisioner_excluded(
         self,
         spec: ExternalResourceSpec,
-        provider_exclusions_by_provisioner_name: Mapping[str, Iterable[str]],
-    ) -> list[str]:
-        return list(
-            provider_exclusions_by_provisioner_name.get(spec.provisioner["name"], [])
-        )
+        provider_exclusions: Mapping[str, Iterable[str]],
+    ) -> bool:
+        exclusions = provider_exclusions.get(spec.provider, [])
+        return "ALL" in exclusions or spec.provisioner_name in exclusions
 
     def _filter_specs_managed_by_erv2(
         self,
         specs: Iterable[ExternalResourceSpec],
-        provider_exclusions_by_provisioner_name: Mapping[str, Iterable[str]],
+        provider_exclusions: Mapping[str, Iterable[str]],
     ) -> list[ExternalResourceSpec]:
-        filtered_specs: list[ExternalResourceSpec] = []
-        for spec in specs:
-            if spec.resource.get("managed_by_erv2"):
-                continue
+        filtered_specs = [
+            spec for spec in specs if not spec.resource.get("managed_by_erv2")
+        ]
 
-            if spec.provider in self._get_provisioner_provider_exclusions(
-                spec, provider_exclusions_by_provisioner_name
-            ):
+        for spec in filtered_specs:
+            if self._is_provisioner_excluded(spec, provider_exclusions):
                 raise ProviderExcludedError(spec)
 
-            filtered_specs.append(spec)
         return filtered_specs
 
+    def _get_provider_exclusions_query_dict(
+        self, provider_exclusions: Iterable[Mapping[str, Any]]
+    ) -> dict[str, list[str]]:
+        exclusions: dict[str, list[str]] = {}
+        for item in provider_exclusions or []:
+            provider = item["provider"]
+            if item.get("excludeAllProvisioners"):
+                exclusions[provider] = ["ALL"]
+            elif item.get("excludeProvisioners"):
+                exclusions[provider] = [
+                    p["name"] for p in item.get("excludeProvisioners") or []
+                ]
```

Comment:
now the structure is more complex than before, better to define a class to hold `excludeAllProvisioners` and `excludeProvisioners`, and avoid magic `ALL`, for example a `Exclusion` dataclass

```python
@dataclass
class Exclusion:
    all = False
    provisioners = field(default_factory=set)
```

then can construct exclusions like

```python
    return {
        item["provider"]: Exclusion(
            all = item.get("excludeAllProvisioners") or False,
            provisioners = {p["name"] for p in item.get("excludeProvisioners") or []}
        )
        for item in provider_exclusions or []
    }
```

check exclude also clearer like

```python
return exclusion.all or spec.provisioner_name in exclusion.provisioners
```

Reply:
yeah, makes sense. Added.